src/lib/wallets/walletConnect/__tests__/walletConnect.test.ts

```diff
@@ -0,0 +1,193 @@
+/**
+ * @jest-environment jsdom
+ */
+import { expect } from '@jest/globals';
+import { Transaction } from 'stellar-sdk';
+
+import {
+    AlreadyRunningError,
+    NoPublicKeyError,
+    NoSessionError,
+    NotRunningError,
+} from '../../../errors/WalletConnectErrors';
+import { WallletConnectService } from '../../../service/walletConnect';
+import LocalStorage from '../../../storage/storage';
+import WalletConnect from '../WalletConnect';
+
+jest.mock('../../../../constants', () => ({
+    PROJECT_ID_FOR_WALLET_CONNECT: 'exampleId',
+    DAPP_BASE_URL: 'exampleUrl',
+}));
+const mockServiceMethods = {
+    connect: jest.fn(),
+    createClient: jest.fn(),
+    makeRequest: jest.fn(),
+    disconnect: jest.fn(),
+};
+jest.mock('../../../service/walletConnect', () => {
+    return {
+        WallletConnectService: jest.fn().mockImplementation(() => {
+            return mockServiceMethods;
+        }),
+        WalletConnectAllowedMethods: jest.fn().mockImplementation(() => {
+            return { SIGN: 'example-method' };
+        }),
+    };
+});
+describe('start', () => {
+    it('should create instance of service', () => {
+        const storage = new LocalStorage();
+        new WalletConnect(storage);
+
+        expect(WallletConnectService).toHaveBeenCalledTimes(1);
+    });
+
+    it('should run the createWalletConnectClient method when wallet starts and throw error if already running', async () => {
+        jest.spyOn(mockServiceMethods, 'createClient').mockResolvedValueOnce({ example: '123' });
+
+        const storage = new LocalStorage();
+        const wallet = new WalletConnect(storage);
+        const startedWallet = await wallet.start();
```

Comment:
Is it possible to declare this outside our tests? That way we don't repeat these lines in every it.